packages/crab-netting/src/CrabNetting.sol

```diff
@@ -7,49 +9,126 @@ contract CrabNetting {
-        crab_balance[msg.sender] = crab_balance[msg.sender] + amount;
-        crab_deposits.push(Receipt(msg.sender, amount));
+    function depositCrab(uint256 _amount) public {
+        IERC20(crab).transferFrom(msg.sender, address(this), _amount);
+        crab_balance[msg.sender] = crab_balance[msg.sender] + _amount;
+        withdraws.push(Receipt(msg.sender, _amount));
     }
 
-    function withdrawCrab(uint256 amount) public {
-        require(crab_balance[msg.sender] >= amount);
-        crab_balance[msg.sender] = crab_balance[msg.sender] - amount;
-        IERC20(crab).transfer(msg.sender, amount);
+    function withdrawCrab(uint256 _amount) public {
+        // TODO ensure final version does not need this check
+        // require(crab_balance[msg.sender] >= _amount);
+        crab_balance[msg.sender] = crab_balance[msg.sender] - _amount;
+        IERC20(crab).transfer(msg.sender, _amount);
     }
 
-    function balanceOf(address account) public view returns (uint256) {
-        return usd_balance[account];
+    function netAtPrice(uint256 _price, uint256 _quantity) public {
+        // TODO make only owner
+        uint256 forWithdraw = _quantity;
+        require(
+            _quantity <= IERC20(usdc).balanceOf(address(this)),
+            "Not enough deposits to net"
+        );
+        // TODO DO we need to do the same check for crab withdrawals?
+        // can we just do deposits alone and forget about withdrawals netting?
+
+        // TODO make sure we dont need SafeMath in final version
+        uint256 i = depositsIndex;
+        while (_quantity > 0 && i < deposits.length) {
+            Receipt memory deposit = deposits[i];
+            if (deposit.amount < _quantity) {
+                _quantity = _quantity - deposit.amount;
+                usd_balance[deposit.sender] =
+                    usd_balance[deposit.sender] -
+                    deposit.amount;
+                IERC20(crab).transfer(deposit.sender, deposit.amount / _price);
+                i++;
+            } else {
+                deposits[i].amount = deposit.amount - _quantity;
+                usd_balance[deposit.sender] =
+                    usd_balance[deposit.sender] -
+                    _quantity;
+                IERC20(crab).transfer(deposit.sender, _quantity / _price);
+                _quantity = 0;
+                delete deposit;
+            }
+        }
+        depositsIndex = depositsIndex + i;
+        uint256 j = withdrawsIndex;
+        uint256 crab_quantity = forWithdraw / _price;
+        while (crab_quantity > 0 && j < withdraws.length) {
+            Receipt memory withdraw = withdraws[j];
+            if (withdraw.amount < crab_quantity) {
+                crab_quantity = crab_quantity - withdraw.amount;
+                crab_balance[withdraw.sender] =
+                    crab_balance[withdraw.sender] -
+                    withdraw.amount;
+                IERC20(usdc).transfer(
```

Comment:
Need events to show PnL/ position to users. 

depositor, withdrawer, usdcAmount, crabAmount